lib/native-client.js

```diff
@@ -104,19 +105,21 @@ class NativeClient extends EventEmitter {
   connect(done) {
     debug('connecting...');
 
+    this.tunnel = null;
```

Comment:
I'm not sure if this can happen - but what if we call `connect(...)` multiple times? Will the old tunnel be left open or should we check if one exists and "silently" close it here? 🤔

Reply:
My understanding is that this case was never explicitly covered by `data-service` because it should be impossible (famous last words, haha) to do something like this. Currently because `connection-model` defines a random port  for the ssh tunnel proxy server on construction, it would immediately throw `EADDRINUSE` if you would call connect multiple times without disconnecting first.

If we want to handle this case I think that maybe we should be more explicit that this is a weird thing to do and throw if we are trying to connect `native-client` a second time. Node does it for their server implementation, calling `server.listen` in sequence will throw. What do you think about this approach?

Reply:
Done in d1d46cd